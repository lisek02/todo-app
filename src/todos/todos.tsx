import * as React from 'react';
import * as R from 'ramda';
import { connect } from 'react-redux';
import { Dispatch, bindActionCreators } from 'redux';
import { TodoList, TodoModel } from './models/todo.model';
import { TodoListComponent } from './components/todoList';
import { TodoFormComponent } from './components/todoForm';
import { addItem, removeItem } from './store/todoListActions';
import { RemoveItemPayload } from './store/todoListActions';

interface StateProps {
  todoList: TodoList
}

interface DispatchProps {
  addItem: (payload: TodoModel) => void;
  removeItem: (payload: RemoveItemPayload) => void;
}

class Component extends React.Component<StateProps & DispatchProps, any>{
  render(){
    return(
      <div className="TodoList">
        <h1>Todo List component</h1>
        <TodoListComponent todos={ this.props.todoList } onRemoveItem={this.props.removeItem} />
        <TodoFormComponent onSubmit={this.props.addItem}/>
      </div>
    );
  }
}

const mapStateToProps = R.applySpec<StateProps>({
  todoList: R.prop('todoList')
});

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(
  {
    addItem,
    removeItem,
  },
  dispatch
);

export const TodosComponent = connect<StateProps, DispatchProps>(mapStateToProps, mapDispatchToProps)(Component);
