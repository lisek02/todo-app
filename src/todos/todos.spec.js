import * as React from 'react';
import { TodosComponent } from './todos';
import { createShallow } from '@material-ui/core/test-utils';
import configureMockStore from 'redux-mock-store';

describe('TodosComponent', () => {
  let shallow;
  const mockStore = configureMockStore([]);
  const store = mockStore({
    todoList: [ { id: 1, title: 'First item'}, { id: 2, title: 'Second item'} ],
  })

  beforeEach(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    store.clearActions();
  });

  
  it('should render correctly with todo as props', () => {
    const component = shallow(<TodosComponent store={store}/>);
    
    expect(component).toMatchSnapshot();
  });
});