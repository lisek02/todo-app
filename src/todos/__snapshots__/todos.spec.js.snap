// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosComponent should render correctly with todo as props 1`] = `
<ContextProvider
  value={null}
>
  <Component
    addItem={[Function]}
    editItem={[Function]}
    removeItem={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    todoList={
      Array [
        Object {
          "id": 1,
          "title": "First item",
        },
        Object {
          "id": 2,
          "title": "Second item",
        },
      ]
    }
    toggleCompleted={[Function]}
    toggleEdit={[Function]}
  />
</ContextProvider>
`;
